docs/doc/60-contributing/03-rfcs/20221028-user-stage.md

```diff
@@ -0,0 +1,133 @@
+---
+title: User Stage
+---
+
+- RFC PR: [datafuselabs/databend#8519](https://github.com/datafuselabs/databend/pull/8519)
+- Tracking Issue: [datafuselabs/databend#8520](https://github.com/datafuselabs/databend/issues/8520)
+
+## Summary
+
+Supports user internal stage.
+
+## Motivation
+
+Databend only supports named internal stage:
+
+```sql
+CREATE STAGE @my_stage;
+COPY INTO my_table FROM @my_stage;
+```
+
+However, named internal stages are complex to be used in some cases. Especially for users who only use stages to load data. By supporting the user stage, they can copy data more efficiently:
+
+```sql
+COPY INTO my_table from @~;
+```
+
+## Guide-level explanation
+
+Databend will add support for user stage. Every sql user will have its own stage, which can be referred to by `~`.
+
+Users can use `~` everywhere like a named stage:
+
+```sql
+COPY INTO my_table FROM @~;
+LIST @~;
+PRESIGN @~/data.csv;
+REMOVE @~ PATTERN = 'ontime.*';
+```
+
+User stage is the internal anonymous stage for sql user, so users can't:
+
+- create
+- drop
+- alter
+
+And users can't set format options for user stage. They need to specify the format settings during `COPY`.
+
+## Reference-level explanation
+
+Databend now has two different [`StageType`](https://github.com/datafuselabs/databend/blob/c2d4e9d3e0a5bf7d54a2a6ce1db1d41b00cd2cd1/src/meta/types/src/user_stage.rs#L52-L55):
+
+```rust
+pub enum StageType {
+    Internal,
+    External,
+}
+```
+
+Databend will generate a unique prefix for the internal stage, like `stage/{stage_name}`.
+
+We will add two new stage types:
+
+```rust
+pub enum StageType {
+    Internal,
+    Named,
+    User,
+    External,
+}
+```
+
+`StageType::Named` will deprecate `StageType::Internal`. Since this RFC, we will not create a new stage with `StageType::Internal` anymore.
+
+The stage prefix rule will be:
+
+- `Internal` => `stage/{stage_name}`
```

Comment:
will this prefix affect the existed stages? q.q

after prefix applied, the storage path might changed

Reply:
oh i see, Internal is for backwards compatibility, how about calls it LegacyInternal ?

Reply:
Nice idea!